packages/flutter/lib/src/material/text_field.dart

```diff
@@ -20,6 +20,15 @@ import 'theme.dart';
 
 export 'package:flutter/services.dart' show TextInputType, TextInputAction, TextCapitalization;
 
+typedef InputCounterWidgetBuilder = Widget Function(
```

Comment:
This typedef needs documentation: explain the purpose of the function and the meaning of each argument.

Typically the first line for typedef doc is boilerplate:
```
/// Signature for the [TextField.buildCounter] callback.
```